src/unix/threadpsx.cpp

```diff
@@ -1734,6 +1744,11 @@ void wxThread::Exit(ExitCode status)
     // might deadlock if, for example, it signals a condition in OnExit() (a
     // common case) while the main thread calls any of functions entering
     // m_critsect on us (almost all of them do)
+    if ( wxSystemOptions::IsFalse("catch-unhandled-exceptions") )
```

Comment:
IMHO it would be nice if we could replace all this section (including the `wxTRY` below) with something like this:

```cpp
CallPossiblyHandlingExceptions([this]() { OnExit(); });
```

where `CallPossiblyHandlingExceptions()` could be a template or just take `std::function<void ()>` (or maybe a template taking `std::function<T ()>` and returning a value of type `T`) defined in the straightforward way.

It's probably not worth to use it for big chunks of code (although maybe it is?), but for one line, or a few line (like above in this file) fragments it looks nicer.

Reply:
I'm not sure how useful a `CallPossiblyHandlingExceptions()` would be in practice. Calling something conditionally with or without exception handling is not a terribly common pattern. It would be used at about 7 places (at most) in code, and after all it would replace pretty simple and readable `if` statements with something less obvious, and only save a few lines of code, total.

Reply:
I dislike having duplicated code, even if it's just a few lines, I've seen too many times just one version of it getting modified later (and it's difficult to see that something that should be modified is _not_, when looking at the diffs during the code review).

And "calling with or without exception handling" looks like a pattern, i.e. something it could be useful to provide a higher level abstraction for.

Note that this could also probably be done with macros, e.g.

```cpp
wxMAYBE_TRY
{
   ... code ...
}
wxMAYBE_CATCH_ALL( ... handler ... );
```

which would be even simpler but probably rather unclear.

Reply:
So thinking more about this, `CallPossiblyHandlingExceptions()` should not only take an `std::function` that it calls, but also another `std::function` for the actions that are to be executed `CATCH_ALL` block (because not all `CATCH_ALL` blocks are identical). And not all `catch` blocks are `CATCH_ALL`, so there would need to be, possibly, a third `std::function` that is called for exceptions of type `abi::__forced_unwind&`. And how nicely does [this](https://github.com/wxWidgets/wxWidgets/blob/c8980d531e83dabff5c278c4e4b35eb13d7fe65f/src/common/evtloopcmn.cpp#L351-L380) turn into a nice little lambda function.

Reply:
It wouldn't have to be a lambda, of course, the same criteria used for using lambdas in other context would apply, i.e. if it's too long, it should be a real function.

Reply:
Lambda or not, I still don't know how could one extract a `try-catch-catch-catch` logic into a function, decoupled from the actual function one wanted to call in the first place.

Since you have a clear vision of what `CallPossiblyHandlingExceptions()` should be like, I think your time is better spent implementing it, rather than trying to explain to me how it should be implemented. I consider this PR to be completed for my part.